packages/@sfx/core/src/core.ts

```diff
@@ -53,8 +62,19 @@ export default class Core {
       throw new Error('Missing dependencies: ' + missingDependencies.join(', '));
     }
 
-    registerPlugins(plugins, this.registry);
+    registerPlugins(plugins, this.registry, this.directory);
+
     initPlugins(plugins);
     readyPlugins(plugins);
   }
+
+  /**
+   * Unregisters all plugins. The plugin registry and directory are both
+   * cleared. The optional `unregister` function of each plugin is
+   * called when the plugin is unregistered. The order in which the
+   * plugins are unregistered is unspecified.
```

Comment:
A dependency graph is the way to do this, but until we decide that it's worth the effort, we decided to leave it unspecified since this approach will work for most cases. If we lock in the order now, we can't change it later.